### ISSUE TITLE
Switch from tut to mdoc

### ISSUE THREAD

**Issue by jdegoes** [scalaz/scalaz-zio#687]
Tut has been great for embedding compiling documentation in our microsite. However, tut has no support for incremental rebuilds and because of its REPL-like approach to code snippets, it means it is difficult to get some kind of code compiling.

This ticket is to convert from `tut` to [Scalameta MDoc](https://scalameta.org/mdoc/), which appears to improve upon `tut` in the two key ways noted above.

**Comment by NeQuissimus** [comment]
Are you looking to replace the microsite entirely? AFAIK tut is embedded into sbt-microsite

**Comment by jdegoes** [comment]
sbt microsite now supports mdoc.

**Comment by NeQuissimus** [comment]
Ah, perfect! Did not know that

**Comment by Dan-M** [comment]
I looked into this one and can provide a PR. But I run into an issue in `testing-effects.md`:

```
lazy val code: ZIO[Database, Throwable, Unit] = ???
lazy val code2: ZIO[Any, Throwable, Unit] = 
  code.provide(TestDatabase)
```
mdoc doesn't allow `???` and fails the build with `NotImplementedError`.
Using `mdoc:crash` still fails the build when accessing `code` in the next statement. I couldn't find a way to get the build work and left this code block without `mdoc` for now.

Another question is if you want to change the directory structure of `microsite` to match the default structure for mdoc?
(I currently just set the `mdocIn := file("microsite/src/main/tut")` so that the changes are easily visible)

...and last but not least: Dependencies for the microsite build have the scope `Tut`. I'm not sure how to  use this for mdoc. So I just removed the scope for now.
